src/base/static/js/views/geometry-editor-view.js

```diff
@@ -0,0 +1,583 @@
+    });
+
+    var generateGeometry = function(layer) {
+      var buildCoords = function(layer) {
+        var coordinates = [],
+        latLngs = layer.getLatLngs();
+        for (var i = 0; i < latLngs.length; i++) {
+          coordinates.push([latLngs[i].lng, latLngs[i].lat]);
+        }
+
+        return coordinates;
+      };
+
+      if (self.geometryType === "polygon" ||
+          self.geometryType === "rectangle" ||
+          self.geometryType === "Polygon") {
+        
+        var coordinates = buildCoords(layer),
+        latLngs = layer.getLatLngs();
+        
+        // Make sure the final polygon vertex exactly matches the first. The
+        // database will reject polygonal geometry otherwise.
+        coordinates.push([latLngs[0].lng, latLngs[0].lat]);
+        self.geometry = {
+          "type": "Polygon",
+          "coordinates": [coordinates]
+        }
+      } else if (self.geometryType === "polyline" ||
+          self.geometryType === "LineString") {
+        
+        var coordinates = buildCoords(layer);
+        self.geometry = {
+          "type": "LineString",
+          "coordinates": coordinates
+        }
+      } else if (self.geometryType === "marker" ||
+          self.geometryType === "Point") {
+
+        self.geometry = {
+          "type": "Point",
+          "coordinates": [layer._latlng.lng, layer._latlng.lat]
+        }
+      }
+    }
+
+    // ========== Drawing events ==========
+    this.map.on("draw:drawvertex", function(evt) {
+      self.numVertices++;
+
+      if (self.layerType === "polygon" && self.numVertices <= 2) {
+        self.displayHelpMessage("draw-polygon-continue-msg");
+      } else if (self.layerType === "polygon" && self.numVertices > 2) {
+        self.displayHelpMessage("draw-polygon-finish-msg");
+      } else if (self.layerType === "polyline" && self.numVertices === 1) {
+        self.displayHelpMessage("draw-polyline-continue-msg");
+      } else if (self.layerType === "polyline" && self.numVertices > 1) {
+        self.displayHelpMessage("draw-polyline-finish-msg");
+      }
+    });
+
+    this.map.on("draw:created", function(evt) {
+      self.resetWorkingGeometry();
+      self.setColorpicker();
+      self.hideIconToolbar();
+
+      self.geometryType = evt.layerType;
+      generateGeometry(evt.layer);
+      self.editingLayerGroup.addLayer(evt.layer);
+      self.swapToolbarVisibility();
+      self.displayHelpMessage("modify-geometry-msg");
+    });
+
+    this.map.on("draw:editvertex", function(evt) {
+      if (self.placeDetailView) {
+        self.placeDetailView.onModified();
+      }
+    });
+
+    this.map.on("draw:editmove", function(evt) {
+      if (self.placeDetailView) {
+        self.placeDetailView.onModified();
+      }
+    });
+
+    this.map.on("draw:edited", function(evt) {
+      evt.layers.eachLayer(function(layer) {
+        
+        // Really there's only one layer to iterate over here, because we have at
```

Comment:
If there were more than one layer to iterate over here, I think what would happen is the geometry editor view would function according to whatever type of geometry the final layer in the iteration is. We could enforce the single layer concept by using `_.once()` I think.